### PR TITLE
Allow explicity specifying which BitmapLoader to use

### PR DESCRIPTION
**Pull request by eygraber**
- There are currently 2 options for using an extension BitmapLoader
    1. Call CropView.extensions().load()
      - This will use whatever image library is available on the
	classpath, starting with Picasso, then Glide, then UIL
      - There can be circumstances where Picasso and Glide are both on the
	classpath, and Glide is the desired library, but Picasso will be
	used instead
    2. Call CropView.extensions().using(GlideBitmapLoader...).load()
      - This will explicitly use the GlideBitmapLoader but loses the
	defered creation of the BitmapLoader if the CropView hasn't been
	attached yet
      - This could be a problem because the size of the Bitmap in
	GlideBitmapLoader is set to CropView's width and height, which if
	it is 0 will cause all images to fail loading

  - This solution provides the best of both worlds; the ability to
    explicitly state which BitmapLoader should be used, and the defered
    creation of it until CropView is attached

### CODE REVIEW

File: scissors-sample/src/main/java/com/lyft/android/scissorssample/MainActivity.java [hunk 1]
```diff
@@ -28,21 +28,19 @@
...
-import com.squareup.leakcanary.RefWatcher;
-
-import java.io.File;
-import java.util.List;
-
 import butterknife.Bind;
 import butterknife.ButterKnife;
 import butterknife.OnClick;
 import butterknife.OnTouch;
+import com.lyft.android.scissors.CropView;
```
Comment (artem-zinnatullin) [inline scissors-sample/src/main/java/com/lyft/android/scissorssample/MainActivity.java:35]: nit: pls don't change imports formatting :)

File: scissors/src/main/java/com/lyft/android/scissors/CropView.java [hunk 2]
```diff
@@ -301,6 +304,28 @@ public LoadRequest using(@Nullable BitmapLoader bitmapLoader) {
             return new LoadRequest(cropView).using(bitmapLoader);
         }
 
+        @IntDef({LOADER_INVALID, LOADER_PICASSO, LOADER_GLIDE, LOADER_UIL})
```
Comment (hlau-lyft) [inline scissors/src/main/java/com/lyft/android/scissors/CropView.java:307]: From API standpoint, Enums are more discoverable and safe for the users of the library, we will merge it as is but will change it to enums in a separate PR :)

File: scissors/src/main/java/com/lyft/android/scissors/CropViewExtensions.java [hunk 3]
```diff
@@ -73,6 +79,17 @@ public LoadRequest using(@Nullable BitmapLoader bitmapLoader) {
         }
 
         /**
+         * Load a {@link Bitmap} using the {@link BitmapLoader} specified by {@code bitmapLoaderReference}, you must call {@link LoadRequest#load(Object)} afterwards.
+         *
+         * @param bitmapLoaderReference a reference to the {@link BitmapLoader} to use
+         * @return current request for chaining, you should call {@link #load(Object)} afterwards.
+         */
+        public LoadRequest using(@CropView.Extensions.ExtensionBitmapLoader int bitmapLoaderReference) {
```
Comment (artem-zinnatullin) [inline scissors/src/main/java/com/lyft/android/scissors/CropViewExtensions.java:87]: 👍

File: scissors/src/main/java/com/lyft/android/scissors/CropViewExtensions.java [hunk 4]
```diff
@@ -73,6 +79,17 @@ public LoadRequest using(@Nullable BitmapLoader bitmapLoader) {
...
+            this.bitmapLoaderReference = bitmapLoaderReference;
+            return this;
```
Comment (artem-zinnatullin) [inline scissors/src/main/java/com/lyft/android/scissors/CropViewExtensions.java:89]: We would need to check for invalid values here to eagerly raise the error, we'll do it in a separate PR :)

**Review by artem-zinnatullin — APPROVED** [review]
👍 Almost a year before merge… sorry about that!

**Review by hlau-lyft — APPROVED** [review]
Sorry, it took so long to merge this.  We had some organizational changes.  Thanks for contributing!
